AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/InitialSync/ModelSyncedEventEmitter.swift

```diff
@@ -0,0 +1,126 @@
+
+        self.modelSyncedEventBuilder = ModelSyncedEvent.Builder()
+
+        self.modelSyncedEventTopic = PassthroughSubject<Never, Never>()
+
+        self.syncOrchestratorSink = initialSyncOrchestrator?
+            .publisher
+            .sink(receiveCompletion: { [weak self] completion in
+                self?.onReceiveCompletion(completion: completion)
+            }, receiveValue: { [weak self] value in
+                self?.onReceiveSyncOperationEvent(value: value)
+            })
+
+        self.reconciliationQueueSink = reconciliationQueue?
+            .publisher
+            .sink(receiveCompletion: { [weak self] completion in
+                self?.onReceiveCompletion(completion: completion)
+            }, receiveValue: { [weak self] value in
+                self?.onReceiveReconciliationEvent(value: value)
+            })
+    }
+
+    private func onReceiveSyncOperationEvent(value: InitialSyncOperationEvent) {
+        switch value {
+        case .isFullSync(let modelType, let isFullSync):
+            guard self.modelType == modelType else {
+                return
+            }
+            modelSyncedEventBuilder.isFullSync = isFullSync
+            modelSyncedEventBuilder.isDeltaSync = !isFullSync
+        case .mutationSync(let mutationSync):
+            guard modelType.modelName == mutationSync.model.modelName else {
+                return
+            }
+            log.info("\(#function): \(value)")
+            _ = recordsReceived.increment()
+        case .finishedOffering(let modelType):
+            guard self.modelType == modelType else {
+                return
+            }
+            initialSyncOpFinished = true
+        }
+    }
+
+    private func onReceiveReconciliationEvent(value: IncomingEventReconciliationQueueEvent) {
+        log.info("\(#function): \(value)")
+        switch value {
+        case .mutationEvent(let event):
+            guard event.modelName == modelType.modelName else {
+                return
+            }
+            _ = reconciledReceived.increment()
+            switch event.mutationType {
+            case "create":
+                _ = modelSyncedEventBuilder.createCount.increment()
+            case "update":
+                _ = modelSyncedEventBuilder.updateCount.increment()
+            case "delete":
+                _ = modelSyncedEventBuilder.deleteCount.increment()
+            default:
+                break
+            }
+            if initialSyncOpFinished == true && reconciledReceived.get() == recordsReceived.get() {
+                dispatchModelSyncedEvent()
+            }
+        case .mutationEventDropped(let name):
+            guard modelType.modelName == name else {
+                return
+            }
+            _ = reconciledReceived.increment()
+        default:
+            return
+        }
+    }
+
+    private func onReceiveCompletion(completion: Subscribers.Completion<DataStoreError>) {
+        log.info("\(#function): \(completion)")
```

Comment:
Oh, do you think `onReceiveCompletion` can be actually removed since it's not very useful? It is not doing anything when a completion is received, so that the sink setup above can be changed to something like
```swift
self.syncOrchestratorSink = initialSyncOrchestrator?
            .publisher
            .sink(receiveCompletion: { _ in },
                  receiveValue: { [weak self] value in
                    self?.onReceiveSyncOperationEvent(value: value)
            })
```
what do you think?

Reply:
yes that looks fine. what's a better place to put logging? perhaps when you dispatch to the Hub?